src/lang.ts

```diff
@@ -40,36 +40,44 @@ interface MixinArgs<T extends {}, U extends {}> {
 	inherited: boolean;
 	sources: (U | null | undefined)[];
 	target: T;
+	copied?: any[];
 }
 
 function _mixin<T extends {}, U extends {}>(kwArgs: MixinArgs<T, U>): T&U {
 	const deep = kwArgs.deep;
 	const inherited = kwArgs.inherited;
-	const target = kwArgs.target;
+	const target: any = kwArgs.target;
+	const copied = kwArgs.copied || [];
 
 	for (let source of kwArgs.sources) {
 		if (source === null || source === undefined) {
 			continue;
 		}
 		for (let key in source) {
 			if (inherited || hasOwnProperty.call(source, key)) {
-				let value: any = (<any> source)[key];
+				let value: any = source[key];
+
+				if (copied.indexOf(value) !== -1) {
```

Comment:
Won't this skip assignment if the same object is referenced by two properties of the source object?

```ts
const a = { foo: 1 };
const b = { one: a, two: a };
const c = deepAssign({}, b);
```